update_workbook.py

```diff
@@ -0,0 +1,249 @@
+
+    return subprocess.run(['git', *args], stdout=subprocess.PIPE, text=True).stdout
+
+
+def git_changed_files(git_base: str = './') -> dict[Path, GitStatus]:
+    git_status = run_git_command('-C', git_base, 'status', '-s')
+    return {
+        Path(file_status[3:]): git_prefix_to_status[status]
+        for file_status in git_status.split('\n')
+        if (status:=file_status[:3]) in git_prefix_to_status
+    }
+
+
+def git_root() -> Path:
+    git_root = run_git_command('rev-parse', '--show-toplevel')
+    return Path(git_root.rstrip())
+
+
+def report_updates() -> None:
+    changed_list = git_changed_files(git_base=str(WORKBOOK_ROOT_DIR))
+
+    if changed_list:
+        print("Updates done.")
+        print("There are updates in the workbooks to be committed.")
+        
+        print("\nHint:\n",
+            "cd workbooks\n",
+            *(f"git add {str(filepath)}\n" for filepath in changed_list),
+            "git commit -m '{COMMITMESSAGE}'\n",
+            "git push origin {BRANCHNAME}")
+    else:
+        print("Nothing to update.")
+
+
+# Helper - System Commands
+def check_current_dir():
+    import os
+    if (cur_file_dir:=Path(__file__).parent) != git_root() \
+        or cur_file_dir != Path(os.getcwd()):
+        raise RuntimeError("`update_workbook` should be run in "
+                           "the `dmsc-school` root directory, where this script is."
+                           f"hint: cd {cur_file_dir.resolve()}")
+
+
+def listmaterials(root_dir: Path) -> list:
+    import os
+    materials = []
+    for lecture_dir in lecture_to_directories.values():
+        try:
+            file_paths = os.listdir(root_dir/lecture_dir)
+            materials.extend([lecture_dir/Path(file_path) for file_path in file_paths])
+        except FileNotFoundError:
+            ...
+    
+    return materials
+
+
+def get_all_materials() -> dict:
+    textbook_list = listmaterials(root_dir=Path('./'))
+    workbook_list = listmaterials(root_dir=WORKBOOK_ROOT_DIR)
+
+    materials = dict()
+    for workbook in workbook_list:
+        if workbook not in textbook_list:
+            materials[workbook] = GitStatus.DELETED
+
+    for textbook in textbook_list:
+        materials[textbook] = GitStatus.TOUCHED
+
+    return materials
+
+
+def export_python_material(textbook_path: Path) -> None:
+    """Copy python materials to the workbook directory."""
+    import shutil
+    shutil.copy(textbook_path, WORKBOOK_ROOT_DIR/textbook_path.parent)
+
+
+# Helper - Jupyter Contents
+def retrieve_tags(cell: dict[str, dict]) -> list:
+    return meta.get('tags', []) if (meta:=cell.get('metadata')) else []
+
+
+def tags_in(cell: dict[str, dict], *tags: str) -> bool:
+    return (cell_tags:=retrieve_tags(cell)) is not None and \
+            all([tag in cell_tags for tag in tags])
+
+
+def is_solution_cell(cell: dict[str, dict]) -> bool:
+    return tags_in(cell, 'solution')
+
+
+def _get_or_make(obj: dict, key: str) -> dict:
```

Comment:
This could be replaced with `dict.setdefault`

Reply:
Right...! I don't think this is used anywhere anymore... I'll delete it...